src/app/util/attribute-storage.cpp

```diff
@@ -1281,8 +1285,15 @@ void emAfSaveAttributeToStorageIfNeeded(uint8_t * data, EndpointId endpoint, Clu
     }
 
     auto * attrStorage = app::GetAttributePersistenceProvider();
-    attrStorage->WriteValue(app::ConcreteAttributePath(endpoint, clusterId, metadata->attributeId), metadata,
-                            ByteSpan(data, dataSize));
+    if (attrStorage)
+    {
+        attrStorage->WriteValue(app::ConcreteAttributePath(endpoint, clusterId, metadata->attributeId), metadata,
+                                ByteSpan(data, dataSize));
+    }
+    else
+    {
+        ChipLogProgress(DataManagement, "Can't store attribute value: no persistence provider");
```

Comment:
This should be an error return, not a log

Reply:
This is a function that returns void...  In practice, we should probably move to crashing if we don't have a provider by the time `emAfLoadAttributeDefaults` happens.

Reply:
Per Slack discussion, leaving as-is for now.